Changes for User Controller and its Unit tests

suprbh opened the pull request:
Modified User Controller file to have Sequelize modifications to access database.
Added partial specs to test the User controller.

@ryojiosawa , @jjmerino , @kmeurer , @jamesongamble - Please review

suprbh committed:
User controller related changes and Specs for tests

suprbh committed:
User controller related changes and Specs for tests

suprbh committed:
Merge branch 'master' of https://github.com/germTheory/germTheoryWebApp

Conflicts:
	specs/server/userModel.spec.js

suprbh committed:
Merge remote-tracking branch 'origin'

ryojiosawa commented on server/controllers/userController.js:
```diff
@@ -4,49 +4,83 @@ var db = require('../database/dbSchema.js'),
+    User.findAll({ where: {name: _username} }).complete(function(err, usrs) {
+                    // attributes: [name, gender] }).complete(function(err, usrs) {
+      if (!!err){
+        console.log('An error occured while finding User: ', _username);
```
@suprbh - I feel findUser and saveUser methods should belong to userModel.js, rather than being in userController.js.  What's the reason that you want to define these methods here?

ryojiosawa commented on server/controllers/userController.js:
```diff
@@ -64,36 +98,42 @@ module.exports = {
+  getUserInfo: function(req, res, next){
+    var username  = req.body.username;
+
+    User.findAll({
```
@suprbh - can't we use findUser method here?  It seems this code is doing the same thing as findUser method.

ryojiosawa commented on server/controllers/userController.js:
```diff
@@ -64,36 +98,42 @@ module.exports = {
+  //   var username  = req.body.username;
+  //   var findUser = Q.nbind(User.findUser, User);
+    
+  //   findUser({username: username})
```
See my comment above.

suprbh committed:
Moved saveUser, findUser from controller to dbSchema.
Modified tests accordingly.
Modified the database connection string to be based on either production of dev. environment.

suprbh committed:
Merge branch 'master' of https://github.com/germTheory/germTheoryWebApp

Conflicts:
	server/controllers/userController.js

suprbh committed:
re-adding file

suprbh committed:
re-adding file

ryojiosawa commented:
Looks good.  Good job!
